WPFHospitalEditor/AdditionalInformation.xaml.cs

```diff
@@ -16,253 +16,114 @@ namespace WPFHospitalEditor
     public partial class AdditionalInformation : Window
     {
         private MapObject mapObject;
-        private List<string> labelContent = new List<string>();
-        private List<string> value = new List<string>();
-        private List<Label> labels = new List<Label>();
-        private List<TextBox> textBoxes = new List<TextBox>();
-        private TextBox name = new TextBox();
-        private int floor;
         private Building building;
         private string[] descriptionParts;
         private string[] contentRows;
         private MapObject oldMapObject;
         private Role role;
         private IEnumerable<EquipmentDto> allEquipment;
+        private DynamicGridControl gridControl;
 
         MapObjectController mapObjectController = new MapObjectController();
         EquipmentServerController equipmentServerController = new EquipmentServerController();
 
-        public AdditionalInformation(MapObject mapObject, Building building, int floor, Role role)
+        public AdditionalInformation(MapObject mapObject, Building building, Role role)
         {
-            this.role = role;
-            this.floor = floor;
-            this.building = building;
-            this.allEquipment = equipmentServerController.getEquipmentByRoomId(mapObject.Id);
             InitializeComponent();
             this.mapObject = mapObject;
-            descriptionParts = mapObject.Description.Split("&");
-            contentRows = descriptionParts[1].Split(";");
-            this.Height = (contentRows.Length + 2) * 50 + 60;
-            ModifyDynamicWPFGrid(contentRows);
-            oldMapObject = mapObject;
+            this.building = building;
+            this.descriptionParts = mapObject.Description.Split("&");
+            this.contentRows = descriptionParts[1].Split(";");
+            this.oldMapObject = mapObject;
+            this.role = role;
+            this.allEquipment = equipmentServerController.getEquipmentByRoomId(mapObject.Id);
+            DynamicGridControl dynamicGridControl = new DynamicGridControl(contentRows, "=", role, false);
+            DynamicGrid.Children.Add(dynamicGridControl);
+            this.gridControl = dynamicGridControl;
+            this.Height = (contentRows.Length + 2) * 50 + 60;       
+            SetButtonsCommonAttributes(Ok);
+            SetButtonsCommonAttributes(Cancel);
+            SetButtonsCommonAttributes(Equipment);
```

Comment:
If you calling these methods in constructor, it is better to define button attributes in xaml.

Reply:
fixed